tasks.py

```diff
@@ -197,47 +190,32 @@ def deploy(space=None, branch=None, yes=False):
     )
     run('cf login {0}'.format(' '.join(' '.join(arg) for arg in args)), echo=True)
 
-    old, new = _detect_apps('api-a', 'api-b')
-
     # Set deploy variables
-    run('cf set-env {0} DEPLOY_BRANCH "{1}"'.format(new, branch))
-    run('cf set-env {0} DEPLOY_USER "{1}"'.format(new, os.getenv('USER')))
-
-    # Push
-    push = run('cf push {0} -f manifest_{1}.yml'.format(new, space), echo=True, warn=True)
-    if push.failed:
-        print('Error pushing app {0}'.format(new))
-        run('cf stop {0}'.format(new), echo=True)
-        return
+    with open('.cfmeta', 'w') as fp:
+        json.dump({'user': os.getenv('USER'), 'branch': branch}, fp)
 
-    # Unset deploy variables
-    run('cf unset-env {0} DEPLOY_BRANCH'.format(new))
-    run('cf unset-env {0} DEPLOY_USER'.format(new))
-
-    # Remap
-    for route, host in SPACE_URLS[space]:
-        opts = route
-        if host:
-            opts += ' -n {0}'.format(host)
-        run('cf map-route {0} {1}'.format(new, opts), echo=True)
-        run('cf unmap-route {0} {1}'.format(old, opts), echo=True, warn=True)
-
-    run('cf stop {0}'.format(old), echo=True, warn=True)
+    # Deploy API
+    run('cf zero-downtime-push api -f manifest_{0}.yml'.format(space), echo=True)
 
     # Deploy worker applications
     run('cf push celery-beat -f manifest_{0}.yml'.format(space))
     run('cf push celery-worker -f manifest_{0}.yml'.format(space))
 
+
 @task
 def notify():
+    try:
+        meta = json.load(open('.cfmeta'))
```

Comment:
Maybe not--the `.cfmeta` file only gets written programmatically, so the only reason to get invalid JSON would be a bug in `deploy`, above. In which case, better to crash than silently fail.